public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -1,234 +1,262 @@
+      allValues.push(d.value);
+    } else {
+      allValues.push(...d.values);
+    }
+  });
+
+  const valueText = options.responseListLength > 1 || options.valueWithRefId ? `Value #${options.refId}` : 'Value';
+
+  const fields: Field[] = [
+    getTimeField(allValues),
+    ...metricLabels.map(label => {
+      return {
+        name: label,
+        config: { filterable: true },
+        type: FieldType.other,
+        values: new ArrayVector(md.map(d => getLabelValue(d.metric, label))),
+      };
+    }),
+    getValueField(allValues, valueText),
+  ];
+
+  return {
+    meta: options.meta,
+    refId: options.refId,
+    length: fields[0].values.length,
+    fields,
+  };
+}
 
-  transformInstantMetricData(md: any, options: any): TimeSeries {
-    const dps = [];
-    const { name, labels } = this.createLabelInfo(md.metric, options);
-    dps.push([parseFloat(md.value[1]), md.value[0] * 1000]);
-    return { target: name ?? '', title: name, datapoints: dps, tags: labels, refId: options.refId, meta: options.meta };
+function getLabelValue(metric: PromMetric, label: string): string | number {
+  if (metric.hasOwnProperty(label)) {
+    if (label === 'le') {
+      return parseHistogramLabel(metric[label]);
+    }
+    return metric[label];
   }
+  return '';
+}
 
-  createLabelInfo(labels: { [key: string]: string }, options: any): { name?: string; labels: Labels; title?: string } {
-    if (options?.legendFormat) {
-      const title = this.renderTemplate(this.templateSrv.replace(options.legendFormat, options?.scopedVars), labels);
-      return { name: title, title, labels };
-    }
+function getTimeField(data: PromValue[], isMs = false): Field {
+  return {
+    name: TIME_SERIES_TIME_FIELD_NAME,
+    type: FieldType.time,
+    config: {},
+    values: new ArrayVector<number>(data.map(val => (isMs ? val[0] : val[0] * 1000))),
+  };
+}
 
-    let { __name__, ...labelsWithoutName } = labels;
+function getValueField(data: PromValue[], valueName: string = TIME_SERIES_VALUE_FIELD_NAME, parseValue = true): Field {
+  return {
+    name: valueName,
+    type: FieldType.number,
+    config: {},
+    values: new ArrayVector<number | null>(data.map(val => (parseValue ? parseFloat(val[1]) : val[1]))),
+  };
+}
 
-    let title = __name__ || '';
+function createLabelInfo(labels: { [key: string]: string }, options: TransformOptions) {
+  if (options?.legendFormat) {
+    const title = renderTemplate(templateSrv.replace(options.legendFormat, options?.scopedVars), labels);
+    return { name: title, labels };
+  }
 
-    const labelPart = formatLabels(labelsWithoutName);
+  let { __name__, ...labelsWithoutName } = labels;
 
-    if (!title && !labelPart) {
-      title = options.query;
-    }
+  let title = __name__ || '';
 
-    title = `${__name__ ?? ''}${labelPart}`;
+  const labelPart = formatLabels(labelsWithoutName);
 
-    return { name: title, title, labels: labelsWithoutName };
+  if (!title && !labelPart) {
+    title = options.query;
```

Comment:
This is a bit weird here, we always overwrite this by the next line so this is a noop, not sure what it should have been doing seems like the same issue was there before.

Reply:
dah, that didn't stick out for me 🤦‍♂️